### DISCUSSION TITLE
Script Tenant

### DISCUSSION THREAD

**Discussion by TomasMondim** [netbox-community/netbox#10976] [General]
Hello, im trying to create an objetcVar for the people to choose the tenant of the device, but i cant find a way to do it like i do in the device_type or in the role!
<img width="381" alt="Capture" src="https://user-images.githubusercontent.com/73384112/202750439-70a079b6-ab7e-45c3-aa68-e55117f8762a.PNG">
It goes well with the others, just fail with the tenant, anyone knows the reason?

**Comment by candlerb** [comment]
What error did you get?

Did you import the tenancy module at the top of your script?

**Reply by TomasMondim** [reply]
how can i had a  location?

**Comment by hagbarddenstore** [comment]
from tenancy.models import Tenant, TenantGroup…tenant_group = ObjectVar(model=TenantGroup, required=False)tenant = ObjectVar(model=Tenant, query_params={”tenant_group_id”: ”$tenant_group”})Sent from my iPhoneOn 18 Nov 2022, at 17:13, Tomas Mondim ***@***.***> wrote:﻿
Hello, im trying to create an objetcVar for the people to choose the tenant of the device, but i cant find a way to do it like i do in the device_type or in the role!

It goes well with the others, just fail with the tenant, anyone knows the reason?

**Reply by candlerb** [reply]
> just fail with the tenant

In what way does it "fail"?

Aside: please edit your post to put three backticks (` ``` `) on a line of their own, before and after the pasted code.  This makes it format properly.

**Reply by candlerb** [reply]
The following script works perfectly for me under Netbox 3.3.8:

```
from tenancy.models import Tenant
from extras.scripts import Script, ObjectVar

class Bar(Script):
    class Meta:
        name = "Bar"
        description = "Testing ObjectVar"

    tenant = ObjectVar(model=Tenant)

    def run(self, data, commit):
        self.log_success(f"Got: {data['tenant']}")
```

It provides a drop-down list of tenants, lets me select one, and gives the name of the tenant in the results.

So you'll have to be more specific about *exactly* what you did that's different, and *exactly* in what way it "fails".

For example: does it produce an exception - if so, what does it say? Does it give an empty list of tenants in the drop-down menu?  Does it allow the user to select a Tenant, but not make it available to the script in data?  etc etc

**Reply by candlerb** [reply]
```
query_params={”tenant_group_id”: ”$tenant_group”})
```

I suggest you remove that first, and see if you get an (unrestricted) list of tenants in the drop-down. If so, you know what you need to investigate further.

Looking at the tenant filter set in the source code [here](https://github.com/netbox-community/netbox/blob/develop/netbox/tenancy/filtersets.py#L144-L157), I think you need:

```
query_params={"group_id": ...
```
or
```
query_params={"group": ...
```
